library/Vanilla/Permissions.php

```diff
@@ -274,6 +293,21 @@ public function setBanned($isBanned) {
     }
 
     /**
+     * Get the flag to determine if all permission checks are being overridden.
+     *
+     * @param bool|null $overrideAll
+     * @return $this
+     */
+    public function setOverrideAll($overrideAll) {
+        if ($overrideAll !== null) {
+            $overrideAll = boolval($overrideAll);
```

Comment:
This is the kind of thing we should stop doing IMO.
The method clearly state that you must receive a boolean or null.

We need, as developers, to ensure that we respect the docblocks.